### PR TITLE
feat(devservices): Support migrations in devenv sync for new devservices

### PR DESCRIPTION
**Pull request by hubertdeng123**
For users using the new devservices, they should set the `USE_NEW_DEVSERVICES` in their .env file. 

devenv sync will bring up new devservices in preparation for migrations. It will also bring down services brought up with `sentry devservices` since those will conflict with new devservices ports.

### CODE REVIEW

**Review by IanWoodard — APPROVED** [review]
LGTM

File: devenv/sync.py [hunk 1]
```diff
@@ -253,18 +255,41 @@ def main(context: dict[str, str]) -> int:
...
-            f"{venv_dir}/bin/{repo}",
-            "devservices",
-            "up",
-            "redis",
-            "postgres",
-        ),
-        pathprepend=f"{reporoot}/.devenv/bin",
-        exit=True,
-    )
+    postgres_container_name = "sentry_postgres"
```
Comment (asottile-sentry) [inline devenv/sync.py:258]: I would put this in the else block -- easier to read and better symmetry

**Review by asottile-sentry — COMMENTED** [review]
hmm I thought the plan was to make `sentry devservices` gracefully use `devservices` when available / used?

**Comment by hubertdeng123** [comment]
> hmm I thought the plan was to make sentry devservices gracefully use devservices when available / used?

That was the plan, but we thought about it some more and decided against it because we wanted people to get into the habit of using`devservices` instead of `sentry devservices`. Additionally, working with click was pretty painful to get it to properly call devservices and pass in the appropriate args.

We can add a warning whenever users use `sentry devservices` if they have the `USE_NEW_DEVSERVICES` set to encourage them to use `devservices` instead for starting dependencies.

@asottile-sentry What do you think?

**Comment by hubertdeng123** [comment]
going to close this PR in favor of another one.

**Pull request closed without merging**